### PR TITLE
OHRM5X-1091: Attendance Update Records

### PR DESCRIPTION
**Pull request by SandeepaOHRM**
This PR contains:

- Working endpoint to update Attendance Records(my and employee)
- Integration testings for Update endpoint
- Working endpoint to get an Attendance Record by id
- Integration testings for getOne endpoint
- Refactoring the UTC identifier into value (+0000)

### CODE REVIEW

File: symfony/plugins/orangehrmCorePlugin/Service/DateTimeHelperService.php [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 class DateTimeHelperService
 {
-    public const TIMEZONE_UTC = 'UTC';
+    public const TIMEZONE_UTC = '+0000';
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmCorePlugin/Service/DateTimeHelperService.php:29]: @samanthajayasinghe need a help here. Used "UTC" according to https://www.php.net/manual/en/timezones.others.php, but it's not working on testing servers. Couldn't found root cause. Do you have any idea.

Requirement: Get calculated UTC time from the server
Expected: Correct UTC time, not based on server timezone
Actual: Getting server time, without converting to UTC. (Not reproducible in dev environment)

```php
$currentDateTime = $this->getDateTimeHelper()->getNow(new DateTimeZone('UTC'));
```

File: symfony/plugins/orangehrmAttendancePlugin/Api/AttendanceRecordAPI.php [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+            $punchInDateTime = $this->extractPunchDateTime($punchInDate.' '.$punchInTime, $punchInTimezoneOffset);
+            $punchInUTCDateTime = (clone $punchInDateTime)->setTimezone(
+                new DateTimeZone(DateTimeHelperService::TIMEZONE_UTC)
+            );
+            $attendanceRecordOwnedEmpNumber = $attendanceRecord->getEmployee()->getEmpNumber();
+            //current state is punched in and editing it
+            if ($attendanceRecord->getState() === AttendanceRecord::STATE_PUNCHED_IN) {
+                $overlappingAttendanceRecords = $this->getAttendanceService()
+                    ->getAttendanceDao()
+                    ->checkForPunchInOverLappingRecordsWhenEditing(
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmAttendancePlugin/Api/AttendanceRecordAPI.php:170]: @vageeshorangehrm shall we refactor methods names,
```suggestion
                    ->checkOverLappingForOpenRecordsWhenEditing(
```

File: symfony/plugins/orangehrmAttendancePlugin/Api/AttendanceRecordAPI.php [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+                $punchOutDateTime = $this->extractPunchDateTime(
+                    $punchOutDate.' '.$punchOutTime,
+                    $punchOutTimezoneOffset
+                );
+                $punchOutUTCDateTime = (clone $punchOutDateTime)->setTimezone(
+                    new DateTimeZone(DateTimeHelperService::TIMEZONE_UTC)
+                );
+                $punchInOverlappingAttendanceRecords = $this->getAttendanceService()
+                    ->getAttendanceDao()
+                    ->checkForPunchInOverLappingRecordsWhenEditing(
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmAttendancePlugin/Api/AttendanceRecordAPI.php:197]: @vageeshorangehrm
```suggestion
                    ->checkOverLappingForClosedRecordsWhenEditing(
```

File: symfony/plugins/orangehrmAttendancePlugin/Api/AttendanceRecordAPI.php [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+    use AuthUserTrait;
+    use UserRoleManagerTrait;
+    use DateTimeHelperTrait;
+
+    public const PARAMETER_PUNCH_IN_DATE = 'punchInDate';
+    public const PARAMETER_PUNCH_IN_TIME = 'punchInTime';
+    public const PARAMETER_PUNCH_IN_NOTE = 'punchInNote';
+    public const PARAMETER_PUNCH_OUT_DATE = 'punchOutDate';
+    public const PARAMETER_PUNCH_OUT_TIME = 'punchOutTime';
+    public const PARAMETER_PUNCH_OUT_NOTE = 'punchOutNote';
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmAttendancePlugin/Api/AttendanceRecordAPI.php:59]: @super-chama can we go with below JSON structure for edit records?
```json
{
  "punchInDate": "",
  "punchInTime": "",
  "punchInNote": "",
  "punchOutDate": "",
  "punchOutTime": "",
  "punchOutNote": "",
}
```

**Review by RajithaKumara — APPROVED** [review]
Good job :tada:

File: symfony/plugins/orangehrmAttendancePlugin/Api/AttendanceRecordAPI.php (hunk 4 above)
Comment (Super-Chama) [reply symfony/plugins/orangehrmAttendancePlugin/Api/AttendanceRecordAPI.php:59]: should be good

**Pull request merged**